zombienet_tests/parachains/0002-parachains-upgrade-smoke-test.feature

```diff
@@ -0,0 +1,11 @@
+Description: Smoke Test
+Network: ./0002-parachains-upgrade-smoke-test.toml
+Creds: config
+
+
+alice: is up
+bob: is up
+alice: parachain 100 is registered within 225 seconds
+alice: parachain 100 block height is at least 10 within 200 seconds
+alice: chain generate dummy upgrade
+alice: parachain 100 block height is at least 14 within 200 seconds
```

Comment:
I am wondering how robust is this?

AFAIU, it may be possible that the upgrade takes more than 4 blocks to be included and this test would succeed even though the upgrade have not taken place.

Reply:
Hi @pepyakin, thanks for the feedback! . We can change the number of `blocks` to wait to be sure that the upgrade is completed, I just set `14` just to ensure that more blocks are produced.  Also when we send the upgrade transaction we wait until is `finalized`.

Thanks!

Reply:
https://github.com/paritytech/cumulus/blob/master/test/service/tests/runtime_upgrade.rs here we have the integration test in Cumulus. We just check that the spec_version increases as expected.

Reply:
I guess the problem will be less likely to happen if we were to increase the number of blocks, but it won't be eliminated. Basti's suggestion is definitely better.

In our case though we probably do not want to bother with the spec_version. What we can do instead, is to verify that the storage entry `:code` becomes updated to the expected value within some limited number of blocks.

It could look like this:

```
alice: chain generate dummy upgrade with hash 0x1337...228
alice: parachain 100 :code becomes 0x1337...228 within 200 seconds
```

Reply:
Hi @pepyakin,
                         I think I can implement two different `upgrade commands`, one that auto-generate the `code` for upgrade like:

```
alice: parachain 100 perform dummy upgrade within 200 seconds
```

This will append the `dummy` code to the actual `:code`, perform the upgrade and validate the `:code` within `x` amount of seconds. Since I think is better to not have to put the `hash` in the `test` files. Also, we can be more explicit and use `... perform dummy upgrade and validate within x seconds`.

The other command is to upgrade to an already generated `code`, providing the `url`. e.g:

```
alice: parachain 100 perform upgrade with <url> within 200 seconds
```
This will  perform the upgrade and validation from the provided `code`.

CC: @ordian / @drahnr

Reply:
Sounds good. The fact that it will check the code hash to verify the upgrade was successful is an implementation detail that doesn't need to surface in the API.